### PR TITLE
Upgrade gosec to v2.9.6

### PR DESCRIPTION
**Pull request by iblancasa**
Signed-off-by: Israel Blancas <iblancas@redhat.com>

## Short description of the changes
- Upgrade `gosec` to use a newer (and more stable) version

### CODE REVIEW

File: pkg/controller/jaeger/account.go [hunk 1]
```diff
@@ -31,7 +31,8 @@ func (r *ReconcileJaeger) applyAccounts(ctx context.Context, jaeger v1.Jaeger, d
 	}
 
 	inv := inventory.ForAccounts(list.Items, desired)
-	for _, d := range inv.Create {
+	for i := range inv.Create {
+		d := inv.Create[i]
```
Comment (frzifus) [inline pkg/controller/jaeger/account.go:35]: why is that needed?

File: pkg/cmd/generate/main.go [hunk 2]
```diff
@@ -81,13 +85,12 @@ func generate(_ *cobra.Command, _ []string) error {
...
 
 	outputName := viper.GetString("output")
-	out, err := os.OpenFile(outputName, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0600)
+	pathToFile := filepath.Clean(outputName)
+	out, err := os.OpenFile(pathToFile, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0600)
 	if err != nil {
 		return err
 	}
 
-	defer out.Close()
```
[the remark this answers is not in the document]
Comment (iblancasa) [reply pkg/cmd/generate/main.go:89]: Hmmm, yes. I'll look for a better solution

File: hack/install/install-gosec.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+# Download the installer
+retry "curl -sLo $BIN/install-gosec.sh https://raw.githubusercontent.com/securego/gosec/master/install.sh"
+chmod +x $BIN/install-gosec.sh
+
+# Run the installer
+retry "$BIN/install-gosec.sh v${VERSION}"
+
+
+export PATH=$PATH:$BIN
```
Comment (rubenvp8510) [inline hack/install/install-gosec.sh:23]: breaking line

File: pkg/controller/jaeger/account.go (hunk 1 above)
Comment (rubenvp8510) [reply pkg/controller/jaeger/account.go:35]: Is gosec marking that here? I understand for others because it takes the address of the loop alias. But I don't get it why is marking this here

File: pkg/util/util.go [hunk 4]
```diff
@@ -4,13 +4,16 @@ import (
...
 	"fmt"
+	"os"
 	"strconv"
 	"strings"
 
 	"github.com/spf13/viper"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
+	"github.com/sirupsen/logrus"
```
Comment (rubenvp8510) [inline pkg/util/util.go:15]: I think you should group this import with the imports above.

**Review by rubenvp8510 — APPROVED** [review]

File: pkg/cmd/generate/main.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (frzifus) [reply pkg/cmd/generate/main.go:89]: Its still a thing, right?
